src/raster/rasterband.rs

```diff
@@ -863,6 +948,82 @@ pub struct StatisticsAll {
     pub std_dev: f64,
 }
 
+#[derive(Debug)]
+pub struct Histogram {
+    min: f64,
+    max: f64,
+    counts: HistogramCounts,
+}
+
+impl Histogram {
+    /// Histogram lower bound
+    pub fn min(&self) -> f64 {
+        self.min
+    }
+
+    /// Histogram upper bound
+    pub fn max(&self) -> f64 {
+        self.max
+    }
+
+    /// Histogram values for each bucket
+    pub fn counts(&self) -> &[u64] {
+        match &self.counts {
```

Comment:
Let's move this to `HistogramCounts::as_slice` and reuse it in the `Debug` implementation.